script/core/definition.lua

```diff
@@ -115,6 +118,11 @@ return function (uri, offset)
     local source = convertIndex(findSource(ast, offset, accept))
     if not source then
         return nil
+    elseif source.type == 'callargs' then
+        source = findSource(ast, offset, acceptGenArg)
```

Comment:
Is these codes really necessary?

Reply:
I believe it is:

This test failes if I comment out these lines:
https://github.com/sumneko/lua-language-server/blob/37cf09ad81a337e4378e39b61a8c52d65e45cfa7/test/crossfile/definition.lua#L80-L89

The idea is as follows: Strings don't usually generate valid references, since they are more often used in other cases than in generic arguments. Hence first get the source as callargs and then check if these contain a generic string. These are two different calls, see `accept` and `acceptGenArg`. If I remember correctly a test case failed, when I tried to just scan for generic strings. I think overshadowed nodes, you actually wanted to search, if not searching for a generic reference.